File: UnitySokoban/Assets/Scripts/Planning/HeuristicSearchPlannerSGW/CompleteSearch.cs
using System;
using System.Collections.Generic;
using Planning;
using StateSpaceSearchProject;

namespace HeuristicSearchPlannerSGW
{
    public class CompleteSearch : HeuristicSearch
    {
        private StateHeuristic heuristic;

        public CompleteSearch(StateSpaceProblem problem, StateHeuristic heuristic, HeuristicComparator comparator)
            :base(problem, heuristic, comparator)
        {
            this.heuristic = heuristic;
            _currentNode = root;
        }

        public override Plan findNextSolution()
        {
            while (queue.size() > 0)
            {
                if (ShutThisSuckaDown)
                {
                    ShutThisSuckaDown = false;
                    return null;
                }

                StateSpaceNode current = queue.pop();
                _currentNode = current;
                current.expand();
                foreach (StateSpaceNode child in current.children)
                    queue.push(child);
                if (problem.goal.IsTrue(current.state))
                    return current.plan;
            }
            return null;
        }

        protected override int GetCost(StateSpaceNode child)
        {
            return Convert.ToInt32(heuristic.evaluate(child.state));
        }
    }
}

File: Heuristic Search (sgware)/src/edu/uno/ai/planning/h/CompleteSearch.java
package edu.uno.ai.planning.h;

import edu.uno.ai.planning.Plan;
import edu.uno.ai.planning.ss.StateSpaceNode;
import edu.uno.ai.planning.ss.StateSpaceProblem;

public class CompleteSearch extends HeuristicSearch {

	public CompleteSearch(StateSpaceProblem problem, StateHeuristic heuristic, HeuristicComparator comparator) {
		super(problem, heuristic, comparator);
	}

	@Override
	public Plan findNextSolution() {
		while(queue.size() > 0) {
			StateSpaceNode current = queue.pop();
			current.expand();
			for(StateSpaceNode child : current.children)
				queue.push(child);
			if(problem.goal.isTrue(current.state))
				return current.plan;
		}
		return null;
	}
}
